### PR TITLE
Add unit tests for parsing browser requests

### PR DESCRIPTION
**Pull request by maximbaz**
Made the functions in `request/process.go` return normal go errors instead of sending messages to the extension, so that it's easier to unit test them.

### CODE REVIEW

File: request/configure.go [hunk 1]
```diff
@@ -11,7 +11,7 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-func configure(request request) {
+func configure(request *request) {
```
Comment (erayd) [inline request/configure.go]: Why the change to a pointer? Efficiency?

File: request/process.go [hunk 2]
```diff
@@ -52,35 +73,21 @@ func Process() {
...
 }
 
-// Request length is the first 4 bytes in LittleEndian encoding on stdin
-func parseRequestLength() uint32 {
+// Request length is the first 4 bytes in LittleEndian encoding
+func parseRequestLength(input io.Reader) (uint32, error) {
 	var length uint32
-	if err := binary.Read(os.Stdin, binary.LittleEndian, &length); err != nil {
-		log.Error("Unable to parse the length of the browser request: ", err)
-		response.SendErrorAndExit(
```
Comment (erayd) [inline request/process.go]: OK, provided this error still gets sent to the client somehow. I'm assuming you intend for this to happen in the caller now.

File: request/configure.go (hunk 1 above)
Comment (maximbaz) [reply request/configure.go]: yeah, I forgot that by default the structures are copied when being passed over as parameters. I noticed this when writing a test :)

File: request/process.go (hunk 2 above)
Comment (maximbaz) [reply request/process.go]: yeah, this code is moved above to the lines 35-55 with no changes

Comment (erayd) [reply request/process.go]: I thought so, but just wanted to double-check your intention was to keep it that way, in light of your comment about changing things so that Go errors were returned instead.

Comment (maximbaz) [reply request/process.go]: I re-read my message and now I see how it is confusing... So confirming: yes, the point is to make this helper function return the normal go error, and make the caller send it to the browser extension.
